doc/yjit/yjit.md

```diff
@@ -499,3 +504,28 @@ PERF=record ruby --yjit-perf=codegen -Iharness-perf benchmarks/lobsters/benchmar
 # Aggregate results
 perf script -s ../ruby/misc/yjit_perf.py
 ```
+
+#### Using perf without Python support
+
+`misc/yjit_perf.py` also has an extra interface to support `perf` without Python support.
+You need to run `perf script` (instead of `perf script -s`) first, and feed the output to `misc/yjit_perf.py`.
+
+This interface exists primarily for production environments. You can run `perf script` in production,
+export the output locally, and run the script using it.
+
+```bash
+# Install perf
+apt-get install linux-tools-common linux-tools-generic linux-tools-`uname -r`
+
+# [Optional] Allow running perf without sudo
+echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+echo -1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+
+# Profile Ruby with --yjit-perf=codegen
+cd ../yjit-bench
+PERF=record ruby --yjit-perf=codegen -Iharness-perf benchmarks/lobsters/benchmark.rb
+
+# Aggregate results
+perf script > /tmp/perf.txt
+../ruby/misc/yjit_perf.py /tmp/perf.txt
```

Comment:
Assuming that this works just as well, maybe we should only give the instructions assuming people are using perf without Python support. Less confusing if we only provide one workflow that works everywhere?

Reply:
Sure, I updated the document that way in https://github.com/ruby/ruby/pull/10246/commits/2abcd19a2456a9bcdefba59daa49977a9cac1471. I still left the instructions to build perf from source in case we want to develop a new feature using something we currently don't parse but is available in the original perf events.